src/main/java/uk/ac/ebi/ampt2d/metadata/persistence/services/LinkedStudyServiceImpl.java

```diff
@@ -0,0 +1,160 @@
+
+            for ( LinkedStudy linkedStudy : linkedStudies ) {
+                Arrays.asList(linkedStudy.getStudy().getId(), linkedStudy.getLinkedStudy().getId())
+                        .forEach(studyId->{
+                            if ( !top.equals(studyId) && !seen.contains(studyId) ) {
+                                ids.add(studyId);
+                                seen.add(studyId);
+                                stack.push(studyId);
+                            }
+                        });
+            }
+        }
+
+        return ids.stream().collect(Collectors.toList());
+    }
+
+    @Override
+    public LinkedStudy findOne(AccessionVersionEntityId studyId, AccessionVersionEntityId linkedStudyId) {
+        Study study = studyRepository.findOne(studyId);
+        Study linkedStudy = studyRepository.findOne(linkedStudyId);
+
+        if ( study == null || linkedStudy == null) {
+            return null;
+        }
+
+        LinkedStudyId linkedStudyId1 = new LinkedStudyId(studyId, linkedStudyId);
+        LinkedStudy linkedStudy1 = linkedStudyRepository.findOne(linkedStudyId1);
+
+        if ( linkedStudy1 != null ) {
+            return linkedStudy1;
+        }
+
+        LinkedStudyId linkedStudyId2 = new LinkedStudyId(linkedStudyId, studyId);
+        LinkedStudy linkedStudy2 = linkedStudyRepository.findOne(linkedStudyId2);
+
+        if ( linkedStudy2 != null ) {
+            return linkedStudy2;
+        }
+
+        return null;
+    }
+
+    @Override
+    public List<Study> findLinkedStudiesById(AccessionVersionEntityId id) {
+        List<AccessionVersionEntityId> ids = findLinkedStudyIds(id);
+
+        QStudy study = QStudy.study;
+        Predicate predicate = study.id.in(ids);
+
+        return (List<Study>) studyRepository.findAll(predicate);
+    }
+
+    @Override
+    public List<LinkedStudy> save(AccessionVersionEntityId id, List<AccessionVersionEntityId> linkedStudyIds) {
+        Study study = studyRepository.findOne(id);
+
+        if ( study == null ) {
+            return Arrays.asList();
+        }
+
+        List<LinkedStudy> linkedStudies = linkedStudyIds.stream().map(linkedStudyId -> {
+            if ( !id.equals(linkedStudyId) ) {
+                Study linkedStudy = studyRepository.findOne(linkedStudyId);
+                if ( linkedStudy != null ) {
+                    return new LinkedStudy(study, linkedStudy);
+                }
+            }
+
+            return null;
+        }).collect(Collectors.toList());
+
+        return (List<LinkedStudy>) linkedStudyRepository.save(linkedStudies);
+    }
+
+    @Override
+    public void delete(AccessionVersionEntityId id) {
+        List<LinkedStudy> linkedStudies = findLinkedStudies(id);
+
+        linkedStudyRepository.delete(linkedStudies);
+    }
+
+    @Override
+    public void delete(AccessionVersionEntityId id, AccessionVersionEntityId linkedStudyId) {
+        LinkedStudy linkedStudy1 = findOne(id, linkedStudyId);
```

Comment:
You can safely do the deletes without the findOne.

Reply:
Can't directly call `delete(ID)` or `delete(T)`. That would cause exception when the item does not exist. However we can use something like `deleteByStudy_IdAndLinkedStudy_Id(studyId, linkedStudyId)`.